test/integration/CalldataExecute.t.sol

```diff
@@ -28,8 +28,126 @@ contract CalldataExecute is Test, Configs {
+            uint256 votingPower = quorumVotes > proposalThreshold
+                ? quorumVotes
+                : proposalThreshold;
+            deal(governanceToken, caller, votingPower);
+
+            // Delegate proposer's votes to itself
+            vm.prank(caller);
+            ERC20Votes(governanceToken).delegate(caller);
+            vm.roll(block.number + 1);
+        }
+
+        uint256 cost = MultichainGovernor(
+            addresses.getAddress("MULTICHAIN_GOVERNOR_PROXY")
+        ).bridgeCostAll();
+
+        vm.deal(address(this), cost);
+        (bool success, bytes memory err) = addresses
+            .getAddress("MULTICHAIN_GOVERNOR_PROXY")
+            .call{value: cost}(execData);
+
+        uint256 proposalId = abi.decode(err, (uint256));
+        require(success, "propose failed");
+
+        /// pass proposal
+        governor.castVote(proposalId, 0);
+
+        // Roll to allow proposal state transitions
+        vm.roll(block.number + governor.votingPeriod() + 1);
+        vm.warp(block.timestamp + governor.votingPeriod() + 1);
+
+        require(
+            governor.state(proposalId) ==
+                IMultichainGovernor.ProposalState.CrossChainVoteCollection,
+            "incorrect state, not succeeded"
+        );
+
+        vm.warp(
+            block.timestamp + governor.crossChainVoteCollectionPeriod() + 1
+        );
+        require(
+            governor.state(proposalId) ==
+                IMultichainGovernor.ProposalState.Succeeded,
+            "incorrect state, not succeeded"
+        );
+
+        deal(governanceToken, governorAddress, 100_000_000 * 1e18);
+        ERC20Votes(governanceToken).balanceOf(governorAddress);
+
+        governor.execute(proposalId);
+    }
+
+    function testSimulateExecMoonriver() public {
+        address caller = address(this);
+        address governanceToken = addresses.getAddress("MFAM");
+        address governorAddress = addresses.getAddress("APOLLO_GOVERNOR");
+        IArtemisGovernor governor = IArtemisGovernor(governorAddress);
+
+        vm.prank(0x6972f25AB3FC425EaF719721f0EBD1Cdb58eE451);
+        ERC20Votes(governanceToken).approve(governorAddress, type(uint256).max);
+
+        deal(governanceToken, governorAddress, 100_000_000 * 1e18);
+
+        {
+            // Ensure proposer has meets minimum proposal threshold and quorum votes to pass the proposal
+            uint256 quorumVotes = governor.currentQuorum();
+            uint256 proposalThreshold = governor.proposalThreshold();
+            uint256 votingPower = quorumVotes > proposalThreshold
+                ? quorumVotes
+                : proposalThreshold;
+            deal(governanceToken, caller, votingPower);
+
+            // Delegate proposer's votes to itself
+            vm.prank(caller);
+            ERC20Votes(governanceToken).delegate(caller);
+            vm.roll(block.number + 1);
+        }
+
+        (bool success, bytes memory err) = address(governorAddress).call(
```

Comment:
I don't mind calling it err, but I personally prefer calling it result since it provides us the proposalId.